### PR TITLE
fix(ACL) update acl actions during user navigation

### PR DESCRIPTION
**Pull request by dmyios**
## Description

This PR addresses the issue of, when administrator changes ACL Actions for a connected user, they are updated in real time during user's page reload.

(Potential issue: admin restricts user ACL Action rights, the user will be having the previous rights until the user's next logout)

The changes were made for Centreon Web Interface ("more actions" on ACL Actions list page & page of editing of ACL Actions) and CLAPI.

**Fixes** # MON-14470

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/class/centreon-clapi/centreonACLAction.class.php [hunk 1]
```diff
@@ -350,4 +354,172 @@ private function exportGrantActions($aclActionRuleId, $aclActionName)
...
+        if (! empty($ids)) {
+            $id = (int) $ids[0];
+            $this->updateAclActionsForAuthentifiedUsers($id);
+            $this->object->delete($id);
+            $this->addAuditLog('d', $id, $objectName);
+            $aclObj = new CentreonACL($this->dependencyInjector);
+            $aclObj->reload(true);
+        } else {
+            throw new CentreonClapiException(self::OBJECT_NOT_FOUND . ":" . $objectName);
+        }
```
Comment (kduret) [inline www/class/centreon-clapi/centreonACLAction.class.php]: use return early pattern : https://medium.com/swlh/return-early-pattern-3d18a41bba8

File: www/class/centreon-clapi/centreonACLAction.class.php [hunk 2]
```diff
@@ -350,4 +354,172 @@ private function exportGrantActions($aclActionRuleId, $aclActionName)
...
+                $statement->execute();
+            }
+        }
+    }
+
+    /**
+     * This function returns user ids from ACL Group Ids
+     *
+     * @param int[] $aclGroupIds
+     * @return int[]
```
Comment (kduret) [inline www/class/centreon-clapi/centreonACLAction.class.php]:
```suggestion
     * @return int[]
     *
     * @throws \InvalidArgumentException
```

File: www/class/centreon-clapi/centreonACLAction.class.php [hunk 3]
```diff
@@ -350,4 +354,174 @@ private function exportGrantActions($aclActionRuleId, $aclActionName)
...
+        } else {
+            throw new CentreonClapiException(self::OBJECT_NOT_FOUND . ":" . $objectName);
+        }
+    }
+
+    /**
+     * @param array $parameters
+     * @throws CentreonClapiException
+     */
+    public function setparam($parameters = array())
```
Comment (dmyios) [inline www/class/centreon-clapi/centreonACLAction.class.php]: Add `: void`

File: www/class/centreon-clapi/centreonACLAction.class.php (hunk 1 above)
Comment (dmyios) [reply www/class/centreon-clapi/centreonACLAction.class.php]: Corrected

**Review by callapa — COMMENTED** [review]
Please use a repository instead of writing SQL queries directly in the CLAPI class.

**Review by kduret — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**